File: ACMICPC/2020 icpc round 1b/local2020Round1BData/solutions/autojudge/autojudge.cpp
#include <iostream>
#include <limits>
#include <string>
#include <vector>

using std::cin;
using std::cout;
using std::endl;
using std::numeric_limits;
using std::streamsize;
using std::string;
using std::vector;

int main()
{
    string probname;
    string infname;
    int i, r, d, e, c, t;
    vector<string> key;
    vector<string> user;

    getline(cin, probname);
    getline(cin, infname);

    cin >> r >> d >> e >> c;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int i = 0; i < c; i++) {
        string line;
        getline(cin, line);
        key.push_back(line);
    }

    cin >> t;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    for (int i = 0; i < t; i++) {
        string line;
        getline(cin, line);
        user.push_back(line);
    }
    
    int dot = infname.rfind('.');
    string ext = infname.substr(dot+1);
    infname = infname.substr(0, dot);

    if (infname != probname || (ext != "c" && ext != "cpp" && ext != "java" && ext != "py")) {
        cout << "Compile Error" << endl;
        return 0;
    }

    if (r) {
        cout << "Run-Time Error" << endl;
        return 0;
    }

    if (e > d) {
        cout << "Time Limit Exceeded" << endl;
        return 0;
    }
    
    bool wa = c != t;
    for (int i = 0; i < c && !wa; i++) {
        wa = key[i] != user[i];
    }

    if (wa) {
        cout << "Wrong Answer" << endl;
        return 0;
    }

    cout << "Correct" << endl; 
}

File: ACMICPC/2020 icpc round 1b/local2020Round1BData/solutions/autojudge/autojudge.java
// Arup Guha
// 9/9/2020
// Alternate Solution for 2020 UCF Locals Round 1B Problem: Autojudge

import java.util.*;

public class autojudge {
	
	// Possible extensions.
	final public static String[] EXT = {".py",".java",".c",".cpp"};

	public static void main(String[] args) {
	
		// Read File stuff.
		Scanner stdin = new Scanner(System.in);
		String fileName = stdin.nextLine();
		String submittedName = stdin.nextLine();
		
		// Read return, timing stuff.
		StringTokenizer tok = new StringTokenizer(stdin.nextLine());
		int ret = Integer.parseInt(tok.nextToken());
		int timeLimit = Integer.parseInt(tok.nextToken());
		int runTime = Integer.parseInt(tok.nextToken());
		
		// Read correct output.
		int numOutLines = Integer.parseInt(stdin.nextLine());
		String[] correct = new String[numOutLines];
		for (int i=0; i<numOutLines; i++)
			correct[i] = stdin.nextLine();
	
		// Read student output.
		int numStudOut = Integer.parseInt(stdin.nextLine());
		String[] student = new String[numStudOut];
		for (int i=0; i<numStudOut; i++)
			student[i] = stdin.nextLine();
		
		// Just go through the cases in the order mentioned.
		if (!matchName(fileName, submittedName))
			System.out.println("Compile Error");
		else if (ret != 0)
			System.out.println("Run-Time Error");
		else if (runTime > timeLimit)
			System.out.println("Time Limit Exceeded");
		else if (!match(correct, student))
			System.out.println("Wrong Answer");
		else
			System.out.println("Correct");
	}
	
	// Returns true if submit is a valid file name for file.
	public static boolean matchName(String file, String submit) {
		
		// To avoid AOOB.
		if (submit.length() < file.length()) return false;
		
		// Match names.
		for (int i=0; i<file.length(); i++)
			if (file.charAt(i) != submit.charAt(i))
				return false;
			
		// Cut off this part.
		submit = submit.substring(file.length());
		
		// Rest must match one of the valid extensions exactly.
		for (int i=0; i<EXT.length; i++)
			if (submit.equals(EXT[i]))
				return true;
			
		// If we get here we never matched a valid extension.
		return false;
	}
	
	// Returns true iff contents of a are identical to b.
	public static boolean match(String[] a, String[] b) {
		
		// Can't be equal if lengths aren't.
		if (a.length != b.length) return false;
		
		// If any line is off, it's bad.
		for (int i=0; i<a.length; i++)
			if (!a[i].equals(b[i]))
				return false;
			
		// We're good.
		return true;
	}
}
